llvm/lib/IR/AsmWriter.cpp

```diff
@@ -1683,8 +1684,28 @@ static const char *getLinkagePrintName(GlobalValue::LinkageTypes LT) {
 
 void MDFieldPrinter::printLinkage(GlobalValue::LinkageTypes LT) {
   Out << FS << "linkage: ";
-  StringRef Linkage = StringRef(getLinkagePrintName(LT)).rtrim(' ');
-  Out << (Linkage.empty() ? "external" : Linkage);
+  const std::string Linkage = getLinkagePrintName(LT);
```

Comment:
Converting to a `std::string` here is a potentially expensive operation. Can we continue to work with `StringRef`?

Reply:
Thanks, Fixed it.